src/lib/utils/handleData.ts

```diff
@@ -0,0 +1,45 @@
+import { IFetchData, IChartDataset, IChartValue } from '@/interface/chartData';
+
+export const transformData = (fetched: IFetchData): IChartDataset => {
+  const copy: IChartDataset = {
+    type: fetched.type,
+    version: fetched.version,
+    value: [],
+  };
```

Comment:
```suggestion
  const copy: IChartDataset = { ...fetched, value: [] };
```
spread문법을 사용해서 간단하게 복사할 수 있습니다!

Reply:
아니요, 그렇지 않습니다. 

매개변수인 fetched의 타입은 아래와 같이 원본 mock데이터의 형태와 똑같습니다. spread하면 `response 속성`의 값 까지 `copy`에 들어가고 타입 에러가 납니다. 

```
export interface IFetchData {
  type: string;
  version: number;
  response: {
    [key: string]: {
      id: string;
      value_area: number;
      value_bar: number;
    };
  };
}
```

또 얕은 복사로 원본 데이터를 참조하게 되어 copy 변수의 사용이 원본 데이터를 오염 시킬 가능성이 있어서 지양해야 한다고 생각합니다. 그래서 `copy` 변수의 타입을 구태여 지정 했었던 것입니다.

Reply:
type과 version만 복사하고 value는 초기화시켜주기 때문에 괜찮을거라 생각했는데 타입 에러가 발생하는군요,, 객체 복사에 대해서 다시한번 공부해봐야겠네요😁